python/pyiceberg/expressions/__init__.py

```diff
@@ -216,7 +216,8 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         return f"And(left={repr(self.left)}, right={repr(self.right)})"
 
-    def __invert__(self) -> Or:
+    def __invert__(self) -> BooleanExpression:
```

Comment:
Minor: Can't we guarantee that the result is an `Or`?

Reply:
It doesn't matter, since the parent class is:
```
class BooleanExpression(ABC):
    """An expression that evaluates to a boolean"""

    @abstractmethod
    def __invert__(self) -> BooleanExpression:
        """Transform the Expression into its negated version."""
```
So we have to check anyway. This is because my PyCharm is complaining:
<img width="770" alt="image" src="https://user-images.githubusercontent.com/1134248/211653685-0d6e6a20-2a70-44a4-a8e4-56e4a8a76b2e.png">